### PR TITLE
Removing SplitQuery plumbing and renaming SplitQueryV2 to SplitQuery

### PR DESCRIPTION
**Pull request by erzel**
@dumbunny @alainjobart

### CODE REVIEW

File: proto/query.proto [hunk 1]
```diff
@@ -508,16 +508,6 @@ message SplitQueryRequest {
...
-
-  // Whether to use the new split-query code
-  // that supports multiple split-columns and
-  // the FULL_SCAN algorithm.
-  // This is a temporary field which aids
-  // in the migration of SplitQuery to the new
-  // code.
-  // TODO(erez): Remove this field after the migration
-  // to the SplitQuery version 2.
-  bool use_split_query_v2 = 10;
```
Comment (dumbunny) [inline proto/query.proto]: Will removing use_split_query_v2 be backwards-compatible with clients?

File: go/cmd/vtgateclienttest/goclienttest/echo.go [hunk 2]
```diff
@@ -426,15 +426,15 @@ func testEchoTransactionExecute(t *testing.T, conn *vtgateconn.VTGateConn) {
 }
 
 func testEchoSplitQuery(t *testing.T, conn *vtgateconn.VTGateConn) {
-	q, err := querytypes.BoundQueryToProto3(echoPrefix+query+":split_column:123", bindVars)
+	q, err := querytypes.BoundQueryToProto3(echoPrefix+query+":{split_column1,split_column2}:123:1000:TODO_BEFORE_SUBMITTING_FILL_IN_ALGORITHM", bindVars)
```
Comment (alainjobart) [inline go/cmd/vtgateclienttest/goclienttest/echo.go]: Just showing I read this TODO_BEFORE_SUBMITTING_FILL_IN_ALGORITHM

File: proto/query.proto (hunk 1 above)
Comment (alainjobart) [inline proto/query.proto]: It should be yes. The clients may specify it or not, the server will just ignore it anyway.

**Comment by erzel** [comment]
Please hold-off on the review. I'm fixing tests. I'll let you know when I'm done.

File: proto/query.proto (hunk 1 above)
Comment (erzel) [inline proto/query.proto]: Right.

**Comment by erzel** [comment]
Ok. @alainjobart @dumbunny This is now ready for review. Thanks!

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**